pycbc/io/hdf.py

```diff
@@ -156,9 +149,10 @@ def select(self, fcn, *args, chunksize=10**6, derived=None, group='',
 
             # Find where it passes the function
             keep = fcn(*partial)
+            remove = np.invert(keep)
 
-            # Keep the indices which pass the function:
-            indices = np.concatenate([indices, submask[keep] + i])
+            # Remove indices which don't pass the function:
+            mask[i:r][remove] = False
```

Comment:
Two things here.

First, would this do anything? Seems like the same kind of construct as the one we decided below wouldn't work.

Second, if it does do something, won't it change `premask` in place, which I don't think we should do.

What is the motivation for keeping the changes to this function?

Reply:
I thought that it would still be an efficiency saving - however it isn't tested to the point where it should be implemented in so deep a change, so I will revert this.

The construct does work here (which I am now confused why the previous one did not):
```
In [1]: import numpy as np

In [2]: mask = np.zeros(10, dtype=bool)

In [3]: mask
Out[3]: 
array([False, False, False, False, False, False, False, False, False,
       False])

In [4]: mask[5:8][2:3] = True

In [5]: mask
Out[5]: 
array([False, False, False, False, False, False, False,  True, False,
       False])

In [6]: np.flatnonzero(mask)
Out[6]: array([7])
```